### ISSUE TITLE
Dialogic 1.x: Random choices are ALL logged in history system

### ISSUE THREAD

**Issue by zaknafean** [dialogic-godot/dialogic#1710]
## The problem

**Describe the bug**
Take a timeline like so:
![image](https://github.com/coppolaemilio/dialogic/assets/7741797/23ebb674-bafc-47f4-a815-282eba0b2b5f)

It runs fine, picks a random choice
![image](https://github.com/coppolaemilio/dialogic/assets/7741797/dd508a3a-2199-4626-835f-2ae88c5ab615)

But  check the history and every choice is logged. Gotta fix that.
![image](https://github.com/coppolaemilio/dialogic/assets/7741797/dc378e0b-aac7-4521-b4ef-f786f205233f)

## Solutions
No solutions or workaround yet. Easy fix is gonna be just make it do the random again, but it would be better if it was consistent.

Any idea to fix the issue or a link to the line of code that might be the cause for this problem

**Comment by Jowan-Spooner** [comment]
I solved this in 2.x by having the "text subsystem" (I know that doesn't exist in 1.x but I think you have a text parser script?) send the finished text to the history system instead of logging it from the event. That's how the whole history works actually: events or subsystems can send "history-info" to the history. 

Not sure if this is at all helpful, just wanted to write it down as we had to deal with the same problem (#1352).
